src/Material/DialogWithCloseButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { Dialog, DialogProps, IconButton, DialogTitle } from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+
+export const DialogWithCloseButton = ({
+	children,
+	onClose,
+	title,
+	...other
+}: Omit<DialogProps, 'title'> & {
+	title: DialogProps['title'] | JSX.Element;
+}) => {
+	return (
+		<Dialog onClose={onClose} {...other}>
+			{title != null && (
+				<DialogTitle
+					sx={{
+						display: 'flex',
+						justifyContent: 'space-between',
+						alignItems: 'flex-start',
+					}}
+					onClick={(e) => {
+						e.preventDefault();
+						e.stopPropagation();
+					}}
+				>
+					{title}
+					{onClose ? (
+						<IconButton
+							aria-label="close"
+							onClick={(e) => {
+								onClose(e, 'backdropClick');
+							}}
+							sx={{
+								color: (theme) => theme.palette.grey[500],
+								ml: 3,
```

Comment:
just to make sure, is this necessary ? I thought the space between rule would be enough ( I might be totally wrong

Reply:
Without it, a long enough title could touch the X: 
![image](https://github.com/balena-io-modules/ui-shared-components/assets/22177187/c189af13-7b96-4034-8bb5-bd870d0d7ae0)